x/pairing/keeper/scores/score.go

```diff
@@ -0,0 +1,230 @@
+//		What is it?
+// 			A Pairing score strategy (or just strategy) defines the weight of each score component in the
+// 			final score calculation for each provider. The calculation is done as follows: req1^w1 + req2^w2 + ...
+//			For example, say that the score components are stake and geolocation. To make the pairing process
+// 			biased towards stake, we could define the strategy: w_stake = 2, w_geo = 1. So the final score
+//			will be providerStakeScore^2 + providerGeoScore^1.
+//		Data structure:
+//			The ScoreStrategy object is a map in which the keys are the requirement object's type and the values
+//			are the requirement's weight.
+//
+//	Adding new requirements:
+// 	  To add a new requirement, one should create a new object that satisfies the ScoreReq interface and update the
+//	  CalcSlots() function so the new requirement will be assigned to the pairing slots (according to some logic).
+//	  Lastly, append the new requirement object to the allReqTypes var in the init() function in this file. Use
+//	  StakeReq or GeoReq as examples.
+
+package scores
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/lavanet/lava/utils"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	planstypes "github.com/lavanet/lava/x/plans/types"
+	tendermintcrypto "github.com/tendermint/tendermint/crypto"
+)
+
+var (
+	uniformStrategy ScoreStrategy
+	allReqNames     []string
+)
+
+// TODO: currently we'll use weight=1 for all reqs. In the future, we'll get it from policy
+func init() {
+	// gather all req names to a list
+	allReqNames = append(allReqNames, stake_req_name)
+
+	// init strategy
+	uniformStrategy = make(ScoreStrategy)
+	for _, reqName := range allReqNames {
+		uniformStrategy[reqName] = 1
+	}
+}
+
+// get the overall requirements from the policy and assign slots that'll fulfil them
+// TODO: this function should be changed in the future since it only supports stake reqs
+func CalcSlots(policy planstypes.Policy, minStake sdk.Int) []*PairingSlot {
+	// init slot array (should be as the number of providers to pair)
+	slots := make([]*PairingSlot, policy.MaxProvidersToPair)
+
+	// all slots should consider the stake, so we init them with stakeReq
+	stakeReq := StakeReq{MinStake: minStake}
+	slotReqs := map[string]ScoreReq{stakeReq.GetName(): stakeReq}
+	for i := range slots {
+		slots[i] = NewPairingSlot(slotReqs)
+	}
+
+	return slots
+}
+
+// group the slots
+func GroupSlots(slots []*PairingSlot) []PairingSlotGroup {
+	slotGroups := []PairingSlotGroup{}
+	if len(slots) == 0 {
+		utils.LavaFormatError("no slots", sdkerrors.ErrLogic)
+		return slotGroups
+	}
+
+	for k := range slots {
+		foundGroup := false
+		for i := range slotGroups {
+			diff := slots[k].Diff(slotGroups[i].Slot)
+			if len(diff.Reqs) == 0 {
+				slotGroups[i].Count += 1
+				foundGroup = true
+				break
+			}
+		}
+
+		if !foundGroup {
```

Comment:
this test is unneeded because in line 125 breaks out of the inner loop when `foundGroup = true`

Reply:
yeah, but if the loop ends without breaking, foundGroup == false and we need to create a new group. This check is necessary

Reply:
right. FWIW, you can use `continue LABEL` (see https://relistan.com/continue-statement-with-labels-in-go), but I see this code was replaced altogether anyhow.